lib/internal/modules/package_json_reader.js

```diff
@@ -1,30 +1,122 @@
 'use strict';
 
-const { SafeMap } = primordials;
+const {
+  JSONParse,
+  ObjectPrototypeHasOwnProperty,
+  SafeMap,
+} = primordials;
+const {
+  ERR_INVALID_PACKAGE_CONFIG,
+} = require('internal/errors').codes;
 const { internalModuleReadJSON } = internalBinding('fs');
-const { pathToFileURL } = require('url');
 const { toNamespacedPath } = require('path');
+const { kEmptyObject } = require('internal/util');
+
+// It is important to require from 'url' and not from 'internal/url'
+// in order to add url module to bootstrapped modules for workers.
+const { fileURLToPath, pathToFileURL } = require('url');
```

Comment:
The justification makes no sense, let's remove `url` from the worker bootstrap path and update the test accordingly (fewer modules in the bootstrap path is a win).
```suggestion
const { fileURLToPath, pathToFileURL } = require('internal/url');
```

Reply:
Removed it. Please review it again.